File: 백준/Silver/2548. 대표 자연수/대표 자연수.py
'''
6
4 3 2 2 9 10
'''

N = int(input())
sorted_arr = sorted(list(map(int, input().split())))

mid = N // 2
minimum_total = int(1e9)
answer = 0
for i in range(mid, -1, -1):
    mid_num = sorted_arr[i]
    total = 0
    
    for n in sorted_arr:
        abs_num = abs(mid_num - n)
        total += abs_num
    
    if total <= minimum_total:
        minimum_total = total
        answer = mid_num
    else:
        break
print(answer)
    

File: 백준/Silver/2548. 대표 자연수/대표 자연수.cc
#include <iostream>
#include <array>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

int main() {
    int m;
    cin >> m;
    
    vector<int> sorted_arr(m);
    
    for (int i = 0; i < m; i++) {
        cin >> sorted_arr[i];
    }
    
    sort(sorted_arr.begin(), sorted_arr.end());

    int minimum_total = 2000000000;
    int answer = 0;

    int mid_num = sorted_arr[m/2];
    
    for (int n = mid_num; n >= 0; n--) {
        int total = 0;
        
        for (int i=0; i < m; i++) {
            int abs_num = abs(sorted_arr[i] - n);
            total += abs_num;
        }

        if (total <= minimum_total) {
            minimum_total = total;
            answer = n;
        }
    }
    cout << answer << endl;
}